client/my-sites/upgrades/map-domain/index.jsx

```diff
@@ -1,136 +1,146 @@
+import wp from 'lib/wp';
+import paths from 'my-sites/upgrades/paths';
 import Notice from 'components/notice';
+import { currentUserHasFlag } from 'state/current-user/selectors';
+import { isSiteUpgradeable } from 'state/selectors';
+import { getSelectedSite, getSelectedSiteId, getSelectedSiteSlug } from 'state/ui/selectors';
+import QueryProductsList from 'components/data/query-products-list';
+
+const wpcom = wp.undocumented();
+
+export class MapDomain extends Component {
+	static propTypes = {
+		initialQuery: PropTypes.string,
+		query: PropTypes.string,
+		cart: PropTypes.object.isRequired,
+		domainsWithPlansOnly: PropTypes.bool.isRequired,
+		isSiteUpgradeable: PropTypes.bool,
+		productsList: PropTypes.object.isRequired,
+		selectedSite: PropTypes.object,
+		selectedSiteId: PropTypes.number,
+		selectedSiteSlug: PropTypes.string,
+		translate: PropTypes.func.isRequired,
+	};
+
+	state = {
+		errorMessage: null
+	};
+
+	goBack = () => {
+		const {
+			selectedSite,
+			selectedSiteSlug,
+		} = this.props;
 
-var MapDomain = React.createClass( {
-	mixins: [ observe( 'productsList', 'sites' ) ],
-
-	propTypes: {
-		query: React.PropTypes.string,
-		productsList: React.PropTypes.object.isRequired,
-		domainsWithPlansOnly: React.PropTypes.bool.isRequired
-	},
-
-	getInitialState: function() {
-		return {
-			errorMessage: null
-		};
-	},
-
-	componentWillMount: function() {
-		this.checkSiteIsUpgradeable();
-	},
-
-	componentDidMount: function() {
-		if ( this.props.sites ) {
-			this.props.sites.on( 'change', this.checkSiteIsUpgradeable );
-		}
-	},
-
-	componentWillUnmount: function() {
-		if ( this.props.sites ) {
-			this.props.sites.off( 'change', this.checkSiteIsUpgradeable );
-		}
-	},
-
-	checkSiteIsUpgradeable: function( ) {
-		if ( ! this.props.sites ) {
-			return;
-		}
-
-		const selectedSite = this.props.sites.getSelectedSite();
-
-		if ( selectedSite && ! selectedSite.isUpgradeable() ) {
-			page.redirect( '/domains/add/mapping' );
-		}
-	},
-
-	goBack: function() {
-		const selectedSite = this.props.sites && this.props.sites.getSelectedSite();
 		if ( ! selectedSite ) {
-			page( this.props.path.replace( '/mapping', '' ) );
+			page( '/domains/add' );
 			return;
 		}
 
 		if ( selectedSite.is_vip ) {
-			page( paths.domainManagementList( selectedSite.slug ) );
+			page( paths.domainManagementList( selectedSiteSlug ) );
 			return;
 		}
 
-		page( '/domains/add/' + selectedSite.slug );
-	},
+		page( '/domains/add/' + selectedSiteSlug );
+	};
+
+	handleRegisterDomain = ( suggestion ) => {
+		const { selectedSiteSlug } = this.props;
 
-	handleRegisterDomain( suggestion ) {
 		upgradesActions.addItem(
 			cartItems.domainRegistration( {
 				productSlug: suggestion.product_slug,
 				domain: suggestion.domain_name
 			} )
 		);
 
-		if ( this.isMounted() ) {
-			page( '/checkout/' + this.props.sites.getSelectedSite().slug );
-		}
-	},
+		page( '/checkout/' + selectedSiteSlug );
```

Comment:
I'm wondering why `isMounted` was added in the first place. Removing it completely can bring some unexpected behavior.